src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponentContext.cs

```diff
@@ -29,32 +42,54 @@ public class ViewComponentContext
         {
             ViewComponentDescriptor = viewComponentDescriptor;
             Arguments = arguments;
-            ViewContext = viewContext;
-            Writer = writer;
+
+            // We want to create a defensive copy of the VDD here so that changes done in the VC
+            // aren't visible in the calling view.
```

Comment:
Nice. This behaviour is also consistent with tag helper activations, HTML helper recursions, and so on.